api/app/api/utils/sql_gen/tests/test_txt_to_sql.py

```diff
@@ -0,0 +1,29 @@
+from typing import Union, OrderedDict, Any, List, Tuple, Callable, Dict
+
+import pytest
+
+from ..text_to_sql import text_to_sql_parallel
+from ...table_selection import get_relevant_tables
+
+Res = Dict[str, Union[List[OrderedDict[str, Any]], List[str]]]
+
+inputs: List[Tuple[str, Callable[[str, Res], str], int]] = [
+    (
+        "What are the three highest income zipcodes in San Jose",
+        [lambda result: any(sub_result.get('zip_code', None) == '95113' for sub_result in result),lambda result: all(sub_result.get('zip_code', None) != '94105' for sub_result in result)],
+    ),
+    (
+        "10 highest crime cities in California",
+        [lambda result: any(sub_result.get('city', '') == 'Los Angeles' for sub_result in result),
+         lambda result: any(sub_result.get('city', '') != 'Los Gatos' for sub_result in result)]
+    )
+]
+
+
+@pytest.mark.parametrize("input_query, expected_output_evaluate_funcs", inputs)
+def test_always_fails(input_query, expected_output_evaluate_funcs):
```

Comment:
this might be going a little overboard and being too general but lmk what y'all think

Reply:
hahaha the los gatos check. looks ok to me